templates/ZF/mnc_5stage/train.prototxt

```
@@ -763,7 +763,7 @@ layer {
 		lr_mult: 2.0 
 	}
 	inner_product_param {
-		num_output: 8
+		num_output: {{ (num_classes|int) * 4 }}
```

This needs to be added in every prototxt template.

---

Is the forced conversion to int really a good idea? The conversion can't fail and will return 0 if the input was garbage. If you don't convert it to int then invalid values may at least cause the prototext parsing to fail instead of silently using 0.

---

That was necessary because `num_classes` was being treated as a string and `num_output` was repeating that value 4 times. So, if `num_classes = 2`, `num_output = 2222` instead of 8. I have fixed that in the Python script, like you proposed.